pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -352,6 +355,7 @@ func (s *DiscoveryServer) removeCon(conID string, con *XdsConnection) {
 		log.Errorf("ADS: Removing connection for non-existing node %v.", s)
 	}
 	delete(adsClients, conID)
+	delete(adsSidecarConfigMap, con.modelNode.ID)
```

Comment:
If the map is only used for debugging - may be ok. But on each envoy restart the 'new' envoy connects before the 'old' one dissconnects - so the entry will be gone. 

One option would be to add without deleting on disconnect - instead have a GC routine that periodically removes
entries without an active connection.  This was a critical optimization in some large scale servers - it saves a lot on
reconnect.